### ISSUE TITLE
Applying a filter to a column with a filter of null throws

### ISSUE THREAD

**Issue by ofirgeller** [AllenFang/react-bootstrap-table#1450]
this is due to this type of checking in 'ableHeaderColumn.js':

``` js
   if (this.props.filter === undefined) {
      return;
    }
```

I recommend using a falsy check:

``` js
   if (!this.props.filter) {
      return;
    }
```

It would catch false/null/undefined.

If you don't check like this and someone used 'null' to mean 'no filter' he is going to get a none specific error when applying a filter. false gives a better error even before you apply due to flow type checks, the problem with null must be that it has the typeof 'object'.

**Comment by AllenFang** [comment]
oh, this is my mistake, I'll fix it and check more details, thanks

**Comment by AllenFang** [comment]
Hello @ofirgeller, this was fixed and release on `v3.4.4`, please have a look on it, thanks
